serialization/aut/src/main/java/net/automatalib/serialization/aut/InternalAUTParser.java

```diff
@@ -187,13 +191,20 @@ private int parseNumberAndShift() {
         final StringBuilder sb = new StringBuilder();
 
         char sym = currentLineContent[currentPos];
+        if(sym == '-') {
+            throw new IllegalArgumentException(buildErrorMessage("Negative numbers not compatible with AUT format"));
+        }
 
         while (Character.isDigit(sym)) {
             sb.append(sym);
             currentPos++;
             sym = currentLineContent[currentPos];
         }
 
+        if(sb.length() == 0){
+            throw new IllegalArgumentException(buildErrorMessage("Expected number"));
+        }
+
```

Comment:
I think the first `if` is not necessary, since a `-` will directly skip the `while` loop and leave the `sb` empty. Your test cases should already cover this. I would just suggest to generalize the error message so something like `Expected a positive number`.

Reply:
Sounds good, changed it!